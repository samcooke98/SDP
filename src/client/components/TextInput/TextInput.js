import React from "react";
import styles from "./web.css";

export default class TextInput extends React.Component {
    render() {
        return (
            <div className={styles.inputContainer} style={this.props.style}>
                <p> {this.props.label} </p>
                <input type={this.props.type} onChange={this.props.onChange} name={this.props.name} value={this.props.value} />
                {this.props.right}
            </div>
        )

    }

    onChange(evt) {
        const name = evt.target.name;
        const value = evt.target.value;
        this.setState({ [name]: value })
    }
}

TextInput.onChange = function(evt) {
    const name = evt.target.name;
    const value = evt.target.value;
    this.setState({ [name]: value })
}

TextInput.defaultProps = { type: "text" }


//In addition to CSS modules, you may wish to use: https://github.com/gajus/react-css-modules