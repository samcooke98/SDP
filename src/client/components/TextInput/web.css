.inputContainer {
    margin-top: 32px;
    margin-bottom: 32px;
}

.inputContainer p {
    margin: 0;
    color: #333333;
    font-family: "Raleway";
    font-weight: bold;
    font-size: 28px;
    margin-bottom: 5px;
}

input {
    width: 100%;
    height: 48px;
    border-radius: 5px;
    border: 1px solid black;
    font-size: 32px;
    padding: 3px 3px;
    box-sizing: border-box;
}