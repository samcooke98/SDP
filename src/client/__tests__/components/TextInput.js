import React from "react";
import renderer from "react-test-renderer";
// import ReactDOM from 'react-dom';

import TextInput from "../../components/TextInput/TextInput";


describe('React Component: TextInput', () => {
    it("matches a snapshot", () => {
        const tree = renderer.create(
            <TextInput />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    })
})