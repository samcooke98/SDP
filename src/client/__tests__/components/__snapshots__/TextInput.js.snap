// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React Component: TextInput matches a snapshot 1`] = `
<div
  className={undefined}
  style={undefined}
>
  <p>
     
     
  </p>
  <input
    autoComplete="off"
    className="undefined undefined"
    name={undefined}
    onChange={undefined}
    placeholder={undefined}
    type="text"
    value={undefined}
  />
</div>
`;
